include/pybind11/cast.h

```diff
@@ -377,6 +377,21 @@ template <typename StringType, bool IsView = false> struct string_caster {
 #endif
         }
 
+#if PY_VERSION_HEX >= 0x03030000
+        // On Python >= 3.3, for UTF-8 we avoid the need for a temporary `bytes`
+        // object by using `PyUnicode_AsUTF8AndSize`.
+        if (UTF_N == 8) {
```

Comment:
```suggestion
        if (PYBIND11_SILENCE_MSVC_C4127(UTF_N == 8)) {
```
I think this is the macro that silences that annoying MSVC warning.

Reply:
Fixed.